File: Frontend/src/app/models/SecuritiesDistribution.ts
export interface SecuritiesDistributionItem {
  id?: number;
  documentType: number;
  serialStart?: string;
  serialEnd?: string;
  count: number;
  price: number;
}

export interface SecuritiesDistributionData {
  id?: number;
  // Tab 1: مشخصات رهنمای معاملات
  registrationNumber: string;
  licenseOwnerName: string;
  licenseOwnerFatherName: string;
  transactionGuideName: string;
  licenseNumber: string;
  // Tab 2: مشخصات اسناد توزیعی
  items: SecuritiesDistributionItem[];
  // Tab 3: قیمت اسناد بهادار
  pricePerDocument?: number;
  totalDocumentsPrice?: number;
  totalSecuritiesPrice?: number;
  // Tab 4: مشخصات آویز تحویلی و تاریخ توزیع
  bankReceiptNumber?: string;
  deliveryDate?: string;
  distributionDate?: string;
  // Formatted dates for display
  deliveryDateFormatted?: string;
  distributionDateFormatted?: string;
  // Legacy fields for backward compatibility with old templates
  documentType?: number;
  propertySubType?: number;
  vehicleSubType?: number;
  propertySaleCount?: number;
  propertySaleSerialStart?: string;
  propertySaleSerialEnd?: string;
  bayWafaCount?: number;
  bayWafaSerialStart?: string;
  bayWafaSerialEnd?: string;
  rentCount?: number;
  rentSerialStart?: string;
  rentSerialEnd?: string;
  vehicleSaleCount?: number;
  vehicleSaleSerialStart?: string;
  vehicleSaleSerialEnd?: string;
  vehicleExchangeCount?: number;
  vehicleExchangeSerialStart?: string;
  vehicleExchangeSerialEnd?: string;
  registrationBookType?: number;
  registrationBookCount?: number;
  registrationBookPrice?: number;
  duplicateBookCount?: number;
  // Audit fields
  createdAt?: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
  status?: boolean;
}

// Alias for backward compatibility
export type SecuritiesDistribution = SecuritiesDistributionData;

// List response interface
export interface SecuritiesDistributionListResponse {
  items: SecuritiesDistributionData[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface DocumentTypeInfo {
  id: number;
  name: string;
  nameEn: string;
  pricePerUnit: number;
  hasSerial: boolean;
}

export const DocumentTypes: DocumentTypeInfo[] = [
  { id: 1, name: 'سټه یی خرید و فروش', nameEn: 'Property Sale', pricePerUnit: 4000, hasSerial: true },
  { id: 2, name: 'سټه یی بیع وفا', nameEn: 'Bay Wafa', pricePerUnit: 4000, hasSerial: true },
  { id: 3, name: 'سټه یی کرایی', nameEn: 'Rent', pricePerUnit: 4000, hasSerial: true },
  { id: 4, name: 'سټه وسایط نقلیه', nameEn: 'Vehicle', pricePerUnit: 4000, hasSerial: true },
  { id: 5, name: 'کتاب ثبت', nameEn: 'Registration Book', pricePerUnit: 1000, hasSerial: false },
  { id: 6, name: 'کتاب ثبت مثنی', nameEn: 'Duplicate Book', pricePerUnit: 20000, hasSerial: false }
];

// Legacy types - kept for backward compatibility but not used in new implementation
export const PropertySubTypes = [
  { id: 1, name: 'سته خرید و فروش' },
  { id: 2, name: 'سته بیع وفا' },
  { id: 3, name: 'سته کرایی' },
  { id: 4, name: 'همه' }
];

export const VehicleSubTypes = [
  { id: 1, name: 'سته خرید و فروش' },
  { id: 2, name: 'سته تبادله' }
];

export const RegistrationBookTypes = [
  { id: 1, name: 'کتاب ثبت' },
  { id: 2, name: 'کتاب ثبت مثنی' }
];

File: Backend/Models/SecuritiesDistribution.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace WebAPIBackend.Models;

/// <summary>
/// اسناد بهادار رهنمای معاملات - Securities Distribution for Transaction Guides
/// </summary>
public partial class SecuritiesDistribution
{
    public int Id { get; set; }

    // Tab 1: مشخصات رهنمای معاملات
    /// <summary>
    /// نمبر ثبت - Registration Number (unique)
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string RegistrationNumber { get; set; } = string.Empty;

    /// <summary>
    /// اسم صاحب امتیاز جواز - License Owner Name
    /// </summary>
    [Required]
    [MaxLength(200)]
    public string LicenseOwnerName { get; set; } = string.Empty;

    /// <summary>
    /// اسم پدر صاحب امتیاز جواز - License Owner Father Name
    /// </summary>
    [Required]
    [MaxLength(200)]
    public string LicenseOwnerFatherName { get; set; } = string.Empty;

    /// <summary>
    /// نام رهنمای معاملات - Transaction Guide Name
    /// </summary>
    [Required]
    [MaxLength(200)]
    public string TransactionGuideName { get; set; } = string.Empty;

    /// <summary>
    /// نمبر جواز - License Number
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string LicenseNumber { get; set; } = string.Empty;

    // Tab 2: مشخصات اسناد توزیعی
    /// <summary>
    /// نوعیت سته - Document Type (1=Property, 2=Vehicle)
    /// </summary>
    public int? DocumentType { get; set; }

    /// <summary>
    /// نوع فرعی سته جایداد - Property Sub Type (1=Sale, 2=BayWafa, 3=Rent, 4=All)
    /// </summary>
    public int? PropertySubType { get; set; }

    /// <summary>
    /// نوع فرعی سته وسایط - Vehicle Sub Type (1=Sale, 2=Exchange)
    /// </summary>
    public int? VehicleSubType { get; set; }

    // Property Document Counts
    /// <summary>
    /// تعداد سته خرید و فروش جایداد
    /// </summary>
    public int? PropertySaleCount { get; set; }

    /// <summary>
    /// سریال نمبر سته خرید و فروش
    /// </summary>
    [MaxLength(100)]
    public string? PropertySaleSerialNumber { get; set; }

    /// <summary>
    /// تعداد سته بیع وفا
    /// </summary>
    public int? BayWafaCount { get; set; }

    /// <summary>
    /// سریال نمبر سته بیع وفا
    /// </summary>
    [MaxLength(100)]
    public string? BayWafaSerialNumber { get; set; }

    /// <summary>
    /// تعداد سته کرایی
    /// </summary>
    public int? RentCount { get; set; }

    /// <summary>
    /// سریال نمبر سته کرایی
    /// </summary>
    [MaxLength(100)]
    public string? RentSerialNumber { get; set; }

    // Vehicle Document Counts
    /// <summary>
    /// تعداد سته خرید و فروش وسایط نقلیه
    /// </summary>
    public int? VehicleSaleCount { get; set; }

    /// <summary>
    /// سریال نمبر سته خرید و فروش وسایط
    /// </summary>
    [MaxLength(100)]
    public string? VehicleSaleSerialNumber { get; set; }

    /// <summary>
    /// تعداد سته تبادله
    /// </summary>
    public int? VehicleExchangeCount { get; set; }

    /// <summary>
    /// سریال نمبر سته تبادله
    /// </summary>
    [MaxLength(100)]
    public string? VehicleExchangeSerialNumber { get; set; }

    // Registration Book
    /// <summary>
    /// نوع کتاب ثبت (1=کتاب ثبت, 2=کتاب ثبت مثنی)
    /// </summary>
    public int? RegistrationBookType { get; set; }

    /// <summary>
    /// تعداد کتاب ثبت
    /// </summary>
    public int? RegistrationBookCount { get; set; }

    /// <summary>
    /// تعداد کتاب ثبت مثنی
    /// </summary>
    public int? DuplicateBookCount { get; set; }

    // Tab 3: قیمت اسناد بهادار
    /// <summary>
    /// قیمت فی جلد سته
    /// </summary>
    [Range(0, double.MaxValue)]
    public decimal? PricePerDocument { get; set; }

    /// <summary>
    /// قیمت مجموعی سته‌ها
    /// </summary>
    [Range(0, double.MaxValue)]
    public decimal? TotalDocumentsPrice { get; set; }

    /// <summary>
    /// قیمت کتاب ثبت / کتاب ثبت مثنی
    /// </summary>
    [Range(0, double.MaxValue)]
    public decimal? RegistrationBookPrice { get; set; }

    /// <summary>
    /// قیمت مجموعی اسناد بهادار
    /// </summary>
    [Range(0, double.MaxValue)]
    public decimal? TotalSecuritiesPrice { get; set; }

    // Tab 4: مشخصات آویز تحویلی و تاریخ توزیع
    /// <summary>
    /// آویز نمبر بانکی
    /// </summary>
    [MaxLength(100)]
    public string? BankReceiptNumber { get; set; }

    /// <summary>
    /// تاریخ تحویلی
    /// </summary>
    public DateOnly? DeliveryDate { get; set; }

    /// <summary>
    /// تاریخ توزیع اسناد بهادار
    /// </summary>
    public DateOnly? DistributionDate { get; set; }

    // Audit Fields
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
    public bool? Status { get; set; }
}
